api/v1alpha1/dnspolicy_types.go

```diff
@@ -69,45 +63,34 @@ type DNSPolicySpec struct {
 	// +optional
 	LoadBalancing *LoadBalancingSpec `json:"loadBalancing,omitempty"`
 
-	// +kubebuilder:validation:Enum=simple;loadbalanced
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="RoutingStrategy is immutable"
-	// +kubebuilder:default=loadbalanced
-	RoutingStrategy RoutingStrategy `json:"routingStrategy"`
-
 	// providerRefs is a list of references to provider secrets. Max is one but intention is to allow this to be more in the future
 	// +kubebuilder:validation:MaxItems=1
 	// +kubebuilder:validation:MinItems=1
 	ProviderRefs []dnsv1alpha1.ProviderRef `json:"providerRefs"`
 }
 
 type LoadBalancingSpec struct {
-	Weighted LoadBalancingWeighted `json:"weighted"`
-
-	Geo LoadBalancingGeo `json:"geo"`
-}
-
-// +kubebuilder:validation:Minimum=0
-type Weight int
-
-type CustomWeight struct {
-	// Label selector to match resource storing custom weight attribute values e.g. kuadrant.io/lb-attribute-custom-weight: AWS.
-	Selector *metav1.LabelSelector `json:"selector"`
-
-	// The weight value to apply when the selector matches.
-	Weight Weight `json:"weight"`
-}
-
-type LoadBalancingWeighted struct {
-	// defaultWeight is the record weight to use when no other can be determined for a dns target cluster.
+	// weight value to apply to weighted endpoints.
 	//
 	// The maximum value accepted is determined by the target dns provider, please refer to the appropriate docs below.
 	//
 	// Route53: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-policy-weighted.html
-	DefaultWeight Weight `json:"defaultWeight"`
+	// Google: https://cloud.google.com/dns/docs/overview/
+	// Azure: https://learn.microsoft.com/en-us/azure/traffic-manager/traffic-manager-routing-methods#weighted-traffic-routing-method
+	Weight int `json:"weight"`
```

Comment:
I think making this optional with a preset default is reasonable rather than forcing it to be set

Reply:
I have no real issue with this ether way tbh. It was made mandatory [here](https://github.com/Kuadrant/kuadrant-operator/pull/578), reason for that as far as i remember was to make it more explicit to users that LoadBalanced is geo weighted, these are provider specific values that you as the user should be aware of, and since you have to set a geo you might as well tell us the weight as well so it's not a surprise where it's coming from.

Since all current providers will support a non negative value and will default to a round robin, equal weighted, when that value whatever it is is the same, setting a default value that will work with all three seems fine to me. "120" seem ok?

Reply:
This is still required in the spec, but now has a default set so you don't have to explicitly set it if you are happy enough using the default value (120)